app/main/routes.py

```diff
@@ -1,22 +1,43 @@
 import requests
 from flask import render_template
 from app.main import bp
-from app.main.forms import GenerateSlangForm
+from app.main.forms import GenerateSlangForm, MeaningForm
 
 
 @bp.route("/", methods=["GET", "POST"])
 @bp.route("/index", methods=["GET", "POST"])
 def index():
-    form = GenerateSlangForm()
+    generate_slang_form = GenerateSlangForm()
+    meaning_form = MeaningForm()
 
-    if form.validate_on_submit():
+    if generate_slang_form.submit_generate.data and generate_slang_form.validate():
         # TODO: Might be overkill to use an API call here rather than internal call.
         response = requests.get("http://localhost:5000/api/generate_slang")
         response = response.json()
         slang_word = response["slang_word"]
+        meaning_form.word.data = slang_word
 
         return render_template(
-            "index.html", title="Home", slang_word=slang_word, form=form
+            "index.html", 
+            title="Home", 
+            slang_word=slang_word, 
+            generate_slang_form=generate_slang_form,
+            meaning_form=meaning_form,
         )
 
-    return render_template("index.html", title="Home", slang_word="", form=form)
+    if meaning_form.submit_meaning.data and meaning_form.validate():
+        print("Successfully submitted the meaning form")
+
+        return render_template(
+            "index.html", 
+            title="Home", 
+            generate_slang_form=generate_slang_form,
+            meaning_form=meaning_form,
+        )
+
+    return render_template(
+        "index.html", 
+        title="Home", 
+        slang_word="", 
+        generate_slang_form=generate_slang_form,
+        meaning_form=meaning_form)
```

Comment:
@annaproxy is it possible to refactor the two title, slang_word, generate_slang_form and meaning_form parameters with **kwargs or smth?

Reply:
Yes, you can make a dictionary beforehand

`kwargs = {'title':'Home', 'meaning_form':meaning_form}`

Then call it twice
```
render_template(slang_word="Hell yes", **kwargs)
render_template(slang_word="", **kwargs)
```

Reply:
Unless the comment intention was "can you please do this" haha then I shall refactor it in a bit

Reply:
Haha, added it in the new commit.